streams/src/main/java/org/apache/kafka/streams/internals/StreamsConfigUtils.java

```diff
@@ -56,12 +56,12 @@ public static ProcessingMode processingMode(final StreamsConfig config) {
         }
     }
 
-    @SuppressWarnings("deprecation")
+    // TODO cleanup
```

Comment:
Same -- follow up PR will refactor this further and remove hard coded string replacements.